src/model/formulations/Gar1962/status.jl

```diff
@@ -2,28 +2,88 @@
 # Copyright (C) 2020, UChicago Argonne, LLC. All rights reserved.
 # Released under the modified BSD license. See COPYING.md for more details.
 
+"""
+    _add_status_vars!
+
+Create `is_on`, `switch_on`, and `switch_off` variables.
+Fix variables if a certain generator _must_ run or based on initial conditions.
+"""
 function _add_status_vars!(
     model::JuMP.Model,
     g::Unit,
     formulation_status_vars::Gar1962.StatusVars,
+    ALWAYS_CREATE_VARS = false,
 )::Nothing
     is_on = _init(model, :is_on)
     switch_on = _init(model, :switch_on)
     switch_off = _init(model, :switch_off)
     for t in 1:model[:instance].time
-        if g.must_run[t]
-            is_on[g.name, t] = 1.0
-            switch_on[g.name, t] = (t == 1 ? 1.0 - _is_initially_on(g) : 0.0)
-            switch_off[g.name, t] = 0.0
-        else
+        if ALWAYS_CREATE_VARS || !g.must_run[t]
```

Comment:
I think the changes to this function made the flow much harder to understand. I suggest the structure below. The point is that only one block `[...]` is executed, so it's easier to follow.
```julia
if always_create_vars
    if g.must_run[t]
        [...]
    else
        [...]
    end
else
    if g.must_run[t]
        [...]
    else
        [...]
    end
end
```

Reply:
You mean instead of

```
            if t == 1
                if _is_initially_on(g)
                    switch_on[g.name, t] = 0.0
                else
                    switch_off[g.name, t] = 0.0
                end
            end
            if g.must_run[t]
                is_on[g.name, t] = 1.0
                switch_on[g.name, t] =
                    (t == 1 ? 1.0 - _is_initially_on(g) : 0.0)
                switch_off[g.name, t] = 0.0
            end
```

you want

```
            if g.must_run[t]
                is_on[g.name, t] = 1.0
                switch_on[g.name, t] =
                    (t == 1 ? 1.0 - _is_initially_on(g) : 0.0)
                switch_off[g.name, t] = 0.0
            elseif t == 1
                if _is_initially_on(g)
                    switch_on[g.name, t] = 0.0
                else
                    switch_off[g.name, t] = 0.0
                end
            end
```

Reply:
I find the second one easier to understand, since I don't have to worry about the case in which both blocks run.

I also just realized that the modified code might be harder to understand because we are replacing more variables by constants; in the original code, only must-run units got replaced, but here we are also replacing some `switch_on` and `switch_off` variables for regular units. This is unexpected to me. It also looks hard to document, and I'm not sure it would have measurable performance impact, so I suggest reverting it. See also comment below about variable replacements in general.